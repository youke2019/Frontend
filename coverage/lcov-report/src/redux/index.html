<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/redux
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.05% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/84</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>36/72</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="69.7" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 69%;"></div><div class="cover-empty" style="width:31%;"></div></div></td>
	<td data-value="69.7" class="pct medium">69.7%</td>
	<td data-value="33" class="abs medium">23/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="11" class="abs low">1/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file low" data-value="reducers.js"><a href="reducers.js.html">reducers.js</a></td>
	<td data-value="24.49" class="pic low"><div class="chart"><div class="cover-fill" style="width: 24%;"></div><div class="cover-empty" style="width:76%;"></div></div></td>
	<td data-value="24.49" class="pct low">24.49%</td>
	<td data-value="49" class="abs low">12/49</td>
	<td data-value="36" class="pct low">36%</td>
	<td data-value="25" class="abs low">9/25</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="48" class="abs low">12/48</td>
	</tr>

<tr>
	<td class="file high" data-value="state.js"><a href="state.js.html">state.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Aug 01 2019 01:16:28 GMT+0800 (GMT+08:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
<script src="..\..\block-navigation.js"></script>
</body>
</html>
