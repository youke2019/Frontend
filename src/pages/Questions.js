import React from "react";
import {
    StyleSheet,
    Text,
    View,
    ImageBackground,
    Image,
    ScrollView,
    TouchableOpacity,
} from "react-native";
import axios from 'axios'
import QuestionCard from '../components/QuestionCard'
import ReplyBox from '../components/ReplyBox'
import {connect} from "react-redux";
import StackNavBar from "../components/StackNavBar";

const mapStateToProps = state => {
    return {
        user: state.user_info
    }
}

class Questions extends React.Component {
    state={
        questions:[],
        question_visible: false,
    }

    componentWillMount() {
        this.flush()
    }

    comeUpQuestion = (data) => {
        axios.post(baseUrl+'/courses/questions/add',{
            course_id: this.props.navigation.state.params.course_info.course_id,
            user_id: this.props.user.id,
            question_content: data
        }).then((res) => {
            this.hideInput()
            this.flush()
        }).catch(err => {
            console.log(err)
        })
    }

    displayInput = () => {
        this.setState({
            question_visible: true
        })
    }

    hideInput = () => {
        this.setState({
            question_visible: false
        })
    }

    flush = () => {
        axios.get(baseUrl+'/courses/questions/find',{
            params:{
                course_id: this.props.navigation.state.params.course_info.course_id,
                user_id: this.props.user.id,
            }
        }).then(res=>{
            this.setState({
                questions: res.data
            })
        }).catch(err=>{
            console.log(err)
        })
    }

    render() {
        return (
            <ScrollView
              style={styles.container}
              keyboardShouldPersistTaps={'handled'}
            >
                <ReplyBox
                    onBackdropPress={this.hideInput}
                    onReplyDone={this.comeUpQuestion}
                    visible={this.state.question_visible}
                />
                <ImageBackground
                    style={styles.header_container}
                    imageStyle={{resizeMode:'stretch'}}
                    source={{uri:'questions_bg'}}
                >
                    <StackNavBar
                        navigation={this.props.navigation}
                    />
                    <View style={styles.title_container}>
                        <Text style={styles.title}>解疑</Text>
                        <View style={styles.subtitle_container}>
                            <Image
                                style={styles.subtitle_icon}
                                source={{uri:'qanda'}}
                            />
                            <Text style={styles.subtitle}>Q&A</Text>
                        </View>
                    </View>
                    <View style={styles.option_container}>
                        <View style={styles.new_question_container}>
                            <TouchableOpacity
                                onPress={this.displayInput}
                            >
                                <ImageBackground
                                    imageStyle={{resizeMode:'stretch'}}
                                    style={styles.new_question}
                                    source={{uri:'black_rectangle'}}
                                >
                                    <Image
                                        style={styles.new_question_icon}
                                        source={{uri:'new_question'}}
                                    />
                                    <Text style={styles.new_question_text}>新的问题</Text>
                                </ImageBackground>
                            </TouchableOpacity>
                        </View>
                    </View>
                </ImageBackground>
                <View style={styles.questions_container}>
                    {
                        this.state.questions.map((item,index) =>
                            <QuestionCard
                                onAnswer={this.flush}
                                QandA = {item}
                                key={index}
                                userId={this.props.user.id}
                            />
                        )
                    }
                </View>
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    container:{
        flex: 1,
        backgroundColor: '#F5F5F5',
    },
    header_container:{
        height:210,
    },
    title_container:{
        flex:1,
        paddingTop: 20,
        paddingHorizontal: 20,
        flexDirection: 'row',
        alignItems: 'center'
    },
    title:{
        color: '#000000',
        fontSize: 50,
        fontFamily: '字魂95号-手刻宋'
    },
    subtitle_container:{
        flexDirection: 'row',
        alignItems: 'center',
        paddingLeft: 10,
    },
    subtitle_icon:{
        width: 30,
        height: 30,
    },
    subtitle:{
        padding: 3,
        color: '#000000',
        fontSize: 12,
    },
    option_container:{
        paddingRight: 30,
        flex:2,
        justifyContent: 'center',
        alignItems: 'flex-end',
    },
    new_question_container:{
        paddingBottom: 60,
    },
    new_question:{
        flexDirection: 'row',
        width: 120,
        height: 40,
        justifyContent: 'center',
        alignItems: 'center',
    },
    new_question_icon:{
        width: 24,
        height: 24,
    },
    new_question_text:{
        paddingTop: 3,
        paddingHorizontal: 3,
        color: '#FFFFFF',
        letterSpacing: 3,
        fontSize: 15,
        fontFamily: '字魂95号-手刻宋',
        lineHeight: 20,
    },
    questions_container:{
        paddingHorizontal:25,
    },
})

export default connect(
    mapStateToProps,
)(Questions)