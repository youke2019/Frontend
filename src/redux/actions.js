export const UPDATE_USER_INFO = 'UPDATEUSERINFO'
export const LOAD_USER_INFO = 'LOADUSERINFO'

export const updateUserInfo = (data) => ({
    type:UPDATE_USER_INFO,
    data
})

export const loadUserInfo = (data) => ({
    type:LOAD_USER_INFO,
    data
})

