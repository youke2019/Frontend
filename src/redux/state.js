const initialState = {
    user_info: null,
    course_list: null,
    login_ready:false,
}

export default initialState