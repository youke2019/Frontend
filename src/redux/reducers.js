import { combineReducers } from 'redux'
import {
    UPDATE_USER_INFO,
    LOAD_USER_INFO,
    CLEAR_USER_INFO,
    SEARCH_COURSES,
} from "./actions.js"
import initialState from "./state";
import axios from 'axios'


function userInfoReducer(state=initialState.user_info, action) {
    switch (action.type) {
        case UPDATE_USER_INFO:
            storage.save({
                key: 'user',
                data: action.data
            })
            state = action.data
            return state
        case LOAD_USER_INFO:
            state = action.data
            return state
        case CLEAR_USER_INFO:
            state = null
            return state
        default:
            return state
    }
}

function courseReducer(state=initialState.course_list, action) {
    switch (action.type) {
        case SEARCH_COURSES:
            state = action.data
            return state
        default:
            return state
    }
}

export const combinedReducer = combineReducers({
    user_info : userInfoReducer,
    course_list: courseReducer,
})