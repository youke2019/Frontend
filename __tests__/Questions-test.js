import React from 'react'
import renderer from 'react-test-renderer'
import {shallow, mount, render, configure} from 'enzyme'
import { createStore } from 'redux'
import { combinedReducer } from '../src/redux/reducers'
import { Provider } from 'react-redux'
import Questions from '../src/pages/Questions'
import Adapter from "enzyme-adapter-react-16/build"

configure({adapter: new Adapter()})

const testInitialState = {
    user_info: {
        id: 'testLzw'
    },
    course_list: null
}

const store = createStore(combinedReducer,testInitialState)

jest.mock('axios')
global.baseUrl = 'baseUrl'

describe('functions', () => {
    let wrapper = shallow(
            <Questions
                store={store}
                navigation={{
                    state:{
                        params:{
                            course_info:{
                                course_id: '123'
                            }
                        }
                    }
                }}
            />
    )

    wrapper = wrapper.dive().dive()

    it('normal refresh()', (done) => {
        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.state('questions')).toHaveLength(1)
                done()
            })
            .catch(err=>{console.log(err)})
    })


    it('hideInput()', () => {
        wrapper.setState({
            question_visible: true
        })
        wrapper.instance().hideInput()
        expect(wrapper.state('question_visible')).toBe(false)
    })

    it('displayInput()', () => {
        wrapper.setState({
            question_visible: false
        })
        wrapper.instance().displayInput()
        expect(wrapper.state('question_visible')).toBe(true)
    })

    it('comeUpQuestion() normal', (done) => {
        wrapper.setState({
            question_visible: true
        })
        wrapper.instance().comeUpQuestion()

        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.state('question_visible')).toBe(false)
                done()
            })
            .catch(err=>{console.log(err)})
    })

    it('comeUpQuestion() abnormal', (done) => {
        wrapper.setState({
            question_visible: true
        })
        wrapper.instance().comeUpQuestion('error')

        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.state('question_visible')).toBe(true)
                done()
            })
            .catch(err=>{console.log(err)})
    })

    it('refresh() error', (done) => {
        global.baseUrl = 'error'
        wrapper.setState({
            questions: []
        })
        wrapper.instance().flush()
        expect(wrapper.state('questions')).toHaveLength(0)

        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                global.baseUrl = 'baseUrl'
                done()
            })
            .catch(err=>{console.log(err)})
    })
})

test('renders QuestionCard page correctly', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <Questions
                navigation={{
                    state:{
                        params:{
                            course_info:{
                                course_id: '123'
                            }
                        }
                    }
                }}
            />
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});