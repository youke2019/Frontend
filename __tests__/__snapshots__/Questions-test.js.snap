// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders QuestionCard page correctly 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="handled"
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
    }
  }
>
  <View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "borderColor": "lightgrey",
            "borderTopWidth": 0.2,
            "bottom": 0,
            "height": 50,
            "paddingHorizontal": 10,
            "paddingVertical": 0,
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="always"
          style={
            Object {
              "contentSize": 0,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "whitesmoke",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderColor": "orange",
                    "borderRadius": 8,
                    "borderWidth": 0.5,
                    "flex": 8,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "marginVertical": 10,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoFocus={true}
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="点击输入回复"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "fontSize": 15,
                      "padding": 2,
                      "textAlign": "left",
                      "width": "100%",
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1.5,
                    "marginVertical": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#FDD326",
                      "borderRadius": 8,
                      "height": "auto",
                      "marginHorizontal": 8,
                      "paddingVertical": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontFamily": "字魂70号-灵悦黑体",
                        "fontSize": 15,
                        "lineHeight": 15,
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                  >
                     发送
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </Modal>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": 210,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "questions_bg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 210,
              "width": undefined,
            },
            Object {
              "resizeMode": "stretch",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginTop": 24,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 6,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "uri": "navigate_return",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 10,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 16,
                "paddingLeft": 8,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            问答
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "paddingHorizontal": 20,
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "字魂95号-手刻宋",
              "fontSize": 50,
            }
          }
        >
          解疑
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingLeft": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "qanda",
              }
            }
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 12,
                "padding": 3,
              }
            }
          >
            Q&A
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 2,
            "justifyContent": "center",
            "paddingRight": 30,
          }
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 60,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                  "width": 120,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "black_rectangle",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 40,
                      "width": 120,
                    },
                    Object {
                      "resizeMode": "stretch",
                    },
                  ]
                }
              />
              <Image
                source={
                  Object {
                    "uri": "new_question",
                  }
                }
                style={
                  Object {
                    "height": 24,
                    "width": 24,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "字魂95号-手刻宋",
                    "fontSize": 15,
                    "letterSpacing": 3,
                    "lineHeight": 20,
                    "paddingHorizontal": 3,
                    "paddingTop": 3,
                  }
                }
              >
                新的问题
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 25,
        }
      }
    />
  </View>
</RCTScrollView>
`;
